### ISSUE TITLE
[BUG] n_words parameter doesn't update y_label values

### ISSUE THREAD

**Issue by ir2718** [MaartenGr/BERTopic#1858]
Hello!

I'm trying to use your library to do simple topic modeling on the Yahoo answers topics dataset. Currently, I'm trying out the unsupervised approaches. When visualizing the final topics, the `n_words` parameter seems to only update the bars inside the plot but not the y label values as well. 

Here is the code and the final image:
```from umap import UMAP
from hdbscan import HDBSCAN
from sentence_transformers import SentenceTransformer
from sklearn.cluster import KMeans
from bertopic.representation import MaximalMarginalRelevance
from bertopic import BERTopic
from bertopic.representation import KeyBERTInspired
from datasets import load_dataset
import matplotlib.pyplot as plt

print("Downloading dataset . . .")
data = load_dataset("yahoo_answers_topics", split="test")
text = data.to_pandas()[["best_answer", "topic"]]
num_classes = len(set(text.topic.values))

print("Downloading model . . .")
embedding_model = SentenceTransformer("sentence-transformers/all-mpnet-base-v2")

umap_model = UMAP(
    n_neighbors=15, n_components=5, min_dist=0.0, metric="cosine"
)
kmeans_model = KMeans(n_clusters=num_classes)

representation_model = [
    KeyBERTInspired(), 
    MaximalMarginalRelevance(diversity=0.5)
]

topic_model = BERTopic(
    embedding_model=embedding_model,
    umap_model=umap_model,
    hdbscan_model=kmeans_model,
    representation_model=representation_model,
    verbose=True,
)

print("Finding topics . . .")
topics, probs = topic_model.fit_transform(text.best_answer.values)

fig = topic_model.visualize_barchart(
    top_n_topics=num_classes,
    n_words=10,
)
fig.write_image("topics_barchart.jpg")
```
![topics_barchart](https://github.com/MaartenGr/BERTopic/assets/94498051/4725b984-6c51-40da-80df-8ea7776952ab)

**Comment by MaartenGr** [comment]
Have you tried increasing the default value of the `height` parameter? It does not show the words because the charts are not long enough, so increasing the value of `height` should do the trick. Other than that, you could also simply zoom in and it will show the words.

**Comment by ir2718** [comment]
This does the trick. I was expecting different default behaviour, as it would be nice to make the font smaller or to change the height automatically.

**Comment by MaartenGr** [comment]
Glad to hear that it worked! It might be possible to have that as default behavior but it might require some complex scaling and exposing Plotly's font (size) in order to calculate how it should be scaled. Either way, happy to hear that it worked and I'll leave this open if somebody wants to work on this automatic scaling. Unfortunately, I cannot take it on at the moment.
